### ISSUE TITLE
Accept NSURL or NSURLCompontents

### ISSUE THREAD

**Issue by jndonald3** [imgix/imgix-swift#7]
It would be convenient if there was a way to build an imgix formatted image from an nsurl.

UseCase:
- Our Api returns full image urls for all entities (i.e. https://domain.com/images123.png?possible_query=stuff). 
  It would be nice to have a function on `ImgixClient` which could somehow take this in as a `NSURL` or `NSURLComponents` object, along with an imgix extra parameters formatted dictionary, and then build the final url preserving the original information from the passed in information.

example:
imgixClient.reconstruct(url:NSURL, params:[String:AnyObject]) -> NSURL?

I have some code which does this in our project, happy to open a PR if you think it's a useful feature.

**Comment by paulstraw** [comment]
Interesting. Just so I'm 100% sure we're on the same page, could you send through example input and output URLs?

**Comment by jndonald3** [comment]
``` swift
let imageUrl = imgixClient.reconstruct(withURL: NSURL(string: "http://images.domain.com/default.png")!, params: ["auto":"format,compress"])
// imageUrl = "http://images.domain.com/default.png?auto=format%2Ccompress&ixlib=swift-0.1.2"
```

**Comment by paulstraw** [comment]
What would happen to the NSURL's query string in this case, if present? Also, would duplicate values be overridden?

**Comment by jndonald3** [comment]
The original query string should be joined with new parameters. 
I'd say that since the developer is supplying the imgix parameter dictionary, its fine to overwrite them from the original query string. 

You could also just provide a flag for that issue to make it more flexible.

**Comment by paulstraw** [comment]
Hey @jndonald3, if possible, would you mind giving #8 a shot and letting me know if that works nicely for you? Thanks!

**Comment by paulstraw** [comment]
Alrighty, released as 0.3.0!
